src/Controller/TideApiController.php

```diff
@@ -343,29 +343,29 @@ private function resolveAliasPath(Request $request, $path, $site, $cid, array &$
 
     // Dispatch a GET_ROUTE event so that other modules can modify it.
     if ($code != Response::HTTP_BAD_REQUEST) {
-      $event_entity = NULL;
-      if ($entity) {
-        $event_entity = clone $entity;
-      }
-      $event = new GetRouteEvent(clone $request, $json_response, $event_entity, $code);
+      $event = new GetRouteEvent(clone $request, $json_response, (!empty($entity) ? clone $entity : NULL), $code);
       $this->eventDispatcher->dispatch(TideApiEvents::GET_ROUTE, $event);
       // Update the response.
       $code = $event->getCode();
       $json_response = $event->getJsonResponse();
-      if ($event->isOk() && $event_entity) {
-        $url = Url::fromRoute('entity.' . $event_entity->getEntityTypeId() . '.canonical', [$event_entity->getEntityTypeId() => $json_response["data"]["entity_id"]]);
-        // Cache the response with the same tags with the entity.
-        $cache_entity = $this->apiHelper->findEntityFromUrl($url);
-        $cached_route_data = [
-          'json_response' => $json_response['data'],
-          'uri' => $url->toUriString(),
-          'id' => $entity ? $entity->uuid() : NULL,
-        ];
+      if ($event->isOk()) {
+        if (!$entity) {
+          // Re-retrieve the entity from the event.
+          if (!empty($json_response['data']['attributes']['entity_id'])) {
+            $entity = $this->entityTypeManager->getStorage($json_response['data']['attributes']['entity_type'])
+              ->load($json_response['data']['attributes']['entity_id']);
+          }
+        }
+
+        if ($entity) {
```

Comment:
@sonnykt Is there a scenario where this conditional wouldn't work as an `else`?
```suggestion
        else {
```

Reply:
without `$entity`, the tide api should not cache the response.

Reply:
And it's not `else` of the previous `if`. The previous `if` attempts to retrieve the entity from the event. The second `if` must run when `$entity` exists, regardless the result of the previous `if`.